### ISSUE TITLE
MHQ Contract with House rights does not generate Allies.

### ISSUE THREAD

**Issue by Thom293** [MegaMek/mekhq#2492]
### Environment
	MekHQ 0.48.0
	Compiled on 2021-03-05T15:43:22Z
	Today is 2021-04-06
	Java vendor Oracle Corporation
	Java version 1.8.0_281
	Platform Windows 10 10.0 (amd64)
	Total memory available to MegaMek: 7,456,768 kB

### Description
I have a contract with House command rights which I think means I get one bot controlled unit per lance.

All other contracts have been normal.  But For some reason this particular contract has never generated an ally.  I even had a base attack and did not get the 2 ally lances, although a spot for them in the mission was generated (you can see it says 0 BV).  The rest of the battles there isnt even a spot generated for them on the individual battles.  I have used the MM force generator to create the Ally teams in the two base attack battles.  Not sure if they will show up but if they do they were manually added.

Screens attached.

Of possible interest.  I do have one custom unit that threw a maint error a few times (its the Enforcer MM).  I used GM to set its quality to F and I think it stopped.  Also this contract was generated with a 160 plus day travel time.  Edit:  After brushing up on my history, I guess the Lyran Alliance split up FedCom in late 3057.  So Fed Com existed when I took the contract, but did not when I landed at the planet (Jan 3058).  I assume the issue is related to that.

Edit 2:  I Edited the mission and changed the employer to Lyran Alliance and it started generating the ally units.  Pretty simple workaround for me the user.  Probably harder for yall though.  Perhaps a check/fix for null employers at the same time the game sometimes recalculates the contract start date?

[The Cartell Customs.zip](https://github.com/MegaMek/mekhq/files/6268002/The.Cartell.Customs.zip)
![ATB no ally](https://user-images.githubusercontent.com/23645569/113789789-734d2b80-9705-11eb-9ec9-bba6668fa1de.png)
![ATB no ally2](https://user-images.githubusercontent.com/23645569/113789812-82cc7480-9705-11eb-80fd-1972a1ea311e.png)
[The Cartell30580404.cpnx.gz](https://github.com/MegaMek/mekhq/files/6267980/The.Cartell30580404.cpnx.gz)
[megameklog.txt](https://github.com/MegaMek/mekhq/files/6267982/megameklog.txt)
[mekhqlog.txt](https://github.com/MegaMek/mekhq/files/6267983/mekhqlog.txt)

**Comment by Windchild292** [comment]
Unrelated Bug:
```
Exception in thread "AWT-EventQueue-0" java.lang.NullPointerException
	at megamek.client.ui.swing.boardview.EntitySprite.getTooltip(EntitySprite.java:1073)
	at megamek.client.ui.swing.boardview.BoardView1.getToolTipText(BoardView1.java:6033)
	at javax.swing.ToolTipManager$insideTimerAction.actionPerformed(Unknown Source)
	at javax.swing.Timer.fireActionPerformed(Unknown Source)
	at javax.swing.Timer$DoPostEvent.run(Unknown Source)
	at java.awt.event.InvocationEvent.dispatch(Unknown Source)
	at java.awt.EventQueue.dispatchEventImpl(Unknown Source)
	at java.awt.EventQueue.access$500(Unknown Source)
	at java.awt.EventQueue$3.run(Unknown Source)
	at java.awt.EventQueue$3.run(Unknown Source)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
	at java.awt.EventQueue.dispatchEvent(Unknown Source)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
	at java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source)
	at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.awt.EventDispatchThread.run(Unknown Source)
```

**Comment by Thom293** [comment]
This is an addition - on the save file I submitted above the Enemy now has invincible morale.

I changed my deployed lances to Overlord and Falcon because Enforcer had someone in the infirmary.  I hit next day.  It generated a Base attack (defense} scenario with Enforcer lance as the chosen lance, even though they were not deployed.  Not sure if that is a bug or feature.  Just wanted to mention.

**Comment by Windchild292** [comment]
Duplicates #251.
As for the enforcer lance... that shouldn't happen, but that might be fixed post-0.48

**Comment by Thom293** [comment]
> Duplicates #251.
> As for the enforcer lance... that shouldn't happen, but that might be fixed post-0.48

Heh.  Funny this is a duplicate of a bug I reported in 2016.  I didnt even remember it until you linked it.
